docs/src/main/paradox/client.md

```diff
@@ -156,7 +156,82 @@ Maven
 mvn akka-grpc:generate compile exec:java -Dexec.mainClass=io.grpc.examples.helloworld.GreeterClient
 ```
 
+### Configuration 
 
+A gRPC client is configured with a `GrpcClientSettings` instance. There are a number of ways of creating one and the API
+docs are the best reference. 
+
+The simplest way to create a client is to provide a static host and port.
+
+Scala
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/scala/docs/akka/grpc/client/GrpcClientSettingsCompileOnly.scala) { #simple }
+
+Java
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/java/jdocs/akka/grpc/client/GrpcClientSettingsCompileOnly.java) { #simple }
+
+Further settings can be added via the `with` methods
+
+Scala
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/scala/docs/akka/grpc/client/GrpcClientSettingsCompileOnly.scala) { #simple-programmatic }
+
+Java
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/java/jdocs/akka/grpc/client/GrpcClientSettingsCompileOnly.java) { #simple-programmatic }
+
+Instead a client can be defined in configuration. All client configurations need to be under `akka.grpc.client`
+
+Scala
+:  @@snip [GrpcClientSettingsSpec]($root$/../runtime/src/test/scala/akka/grpc/GrpcClientSettingsSpec.scala) { #client-config }
+
+Java
+:  @@snip [GrpcClientSettingsSpec]($root$/../runtime/src/test/scala/akka/grpc/GrpcClientSettingsSpec.scala) { #client-config }
+
+Clients defined in configuration pick up defaults from `reference.conf`:
+
+Scala
+:  @@snip [reference]($root$/../runtime/src/main/resources/reference.conf) { #defaults }
+
+Java
+:  @@snip [reference]($root$/../runtime/src/main/resources/reference.conf) { #defaults }
+
+#### Service discovery
+
+The examples above all use a hard coded host and port for the location of the gRPC service. Alternatively 
```

Comment:
" which is the default if you do not configure a `service-discovery-mechanism`." or something like that?